scipy/stats/_continuous_distns.py

```diff
@@ -8426,6 +8433,17 @@ def _stats(self, beta):
     def _entropy(self, beta):
         return 1. / beta - np.log(.5 * beta) + sc.gammaln(1. / beta)
 
+    def _rvs(self, beta, size=None, random_state=None):
+        # see [2]_ for the algorithm
+        # see [3]_ for reference implementation in SAS
+        z = random_state.gamma(1/beta, size=size)
+        y = z ** (1/beta)
+        # convert y to array to ensure masking support
+        y = np.asarray(y)
+        mask = random_state.random(size=y.shape) < 0.5
+        y[mask] = -y[mask]
+        return y
```

Comment:
I haven't checked out the branch. Is there something in `rv_continuous.rvs` that protects us from returning a 0d array here?

Reply:
Apparantly this line:

![image](https://user-images.githubusercontent.com/43181252/149611536-1fd7b607-7621-4217-9d96-5957a9fb8378.png)

casts a `np.ndarray` into a `np.float64` because `_parse_args_rvs` makes sure that `loc` and `scale` are `np.float64` if scalar values are passed. This is kind-of hacky. If you want, I can change the return line to `return y[()]` to make sure a float64 is returned by `_rvs` if a scalar value is sampled.

Reply:
OK, so a 0D array times a float is a float. I see. 
It's fine as-is, but whenever we rewrite this stuff, we'll want to be more explicit so nobody has to guess in the future.